registry/remote/example_test.go

```diff
@@ -0,0 +1,539 @@
+	}
+	fmt.Println(string(pulled))
+
+	// Output:
+	// Example manifest content
+}
+
+// ExampleRepository_FetchReference_byTag gives example snippets for downloading a manifest by tag with only one API call.
+func ExampleRepository_FetchReference_manifestByTag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	tag := "latest"
+	reg.ManifestMediaTypes = append(reg.ManifestMediaTypes, ocispec.MediaTypeImageManifest) // Specify the expected manifest media type
+	descriptor, r, err := repo.FetchReference(ctx, tag)                                     // Fetch the blob from the repository
+	if err != nil {
+		panic(err) // Handle error
+	}
+	defer r.Close() // don't forget to close
+	pulledBlob, err := io.ReadAll(r)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(string(pulledBlob))
+	// caller SHOULD verify what is fetched
+	if descriptor.Digest != ocidigest.FromBytes(pulledBlob) || descriptor.Size != int64(len(pulledBlob)) {
+		panic(err) // Handle error
+	}
+
+	// Output:
+	// Example manifest content
+}
+
+// ExampleRepository_FetchReference_byDigest gives example snippets for downloading a manifest by digest.
+func ExampleRepository_FetchReference_manifestByDigest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	digest := "sha256:00e5ffa7d914b4e6aa3f1a324f37df0625ccc400be333deea5ecaa199f9eff5b"
+	descriptor, r, err := repo.FetchReference(ctx, digest) // Fetch the blob from the repository
+	if err != nil {
+		panic(err) // Handle error
+	}
+	defer r.Close() // don't forget to close
+	pulled, err := io.ReadAll(r)
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	// caller SHOULD verify what is fetched
+	if descriptor.Digest != ocidigest.FromBytes(pulled) || descriptor.Size != int64(len(pulled)) {
+		panic(err) // Handle error
+	}
+	fmt.Println(string(pulled))
+
+	// Output:
+	// Example manifest content
+}
+
+// ExampleRepository_Fetch_layer gives example snippets for downloading a layer blob by digest.
+func ExampleRepository_Fetch_layer() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	descriptor := ocispec.Descriptor{
+		MediaType: ocispec.MediaTypeImageLayer,
+		Digest:    digest.FromBytes([]byte(exampleLayer)),
+		Size:      int64(len(exampleLayer)),
+	}
+	r, err := repo.Fetch(ctx, descriptor) // Fetch the blob from the repository
```

Comment:
You may want to use `Repository.Blobs().Resolve()` to get the descriptor of the layer? See [Test_BlobStore_Resolve](https://github.com/oras-project/oras-go/blob/main/registry/remote/repository_test.go#L1421-L1424) and [Test_BlobStore_FetchReference](https://github.com/oras-project/oras-go/blob/main/registry/remote/repository_test.go#L1495-L1499).